### ISSUE TITLE
POZYX_UWB_GAIN register

### ISSUE THREAD

**Issue by lpdunwell** [pozyxLabs/Pozyx-Arduino-library#22]
I noticed setTxPower and getTxPower both use floats, however in UWB_settings_t gain is an int8_t.
setTxPower and setUWBSettings don't work as intended.
Running this code:
```
#include <Pozyx.h>
#include <Pozyx_definitions.h>
#include <Wire.h>

void setup()
{
  int status;
  float gain = 0;
  UWB_settings_t settings;
  
  Serial.begin(115200);

  Pozyx.begin(true);

  Serial.println("\nDefault values:");
  Pozyx.getUWBSettings(&settings);
  Serial.print("getUWBSettings.gain: ");
  Serial.println(settings.gain, DEC);
  Pozyx.getTxPower(&gain);
  Serial.print("getTxPower: ");
  Serial.println(gain, DEC);

  Serial.println("\nStarting setTxPower test:");

  for(int i = 0; i < 36; i++)
  {
    Serial.print("setTxPower:\t\t\t");
    Serial.println(i);
    Pozyx.setTxPower(i);
    Pozyx.getUWBSettings(&settings);
    Serial.print("getUWBSettings.gain:\t\t");
    Serial.println(settings.gain, DEC);
    Pozyx.getTxPower(&gain);
    Serial.print("getTxPower:\t\t\t");
    Serial.println(gain, DEC);
    Serial.println();
  }

  Serial.println("\n\nStarting setUWBSettings test:");

  for(int i = 0; i < 71; i++)
  {
    Serial.print("setUWBSettings:\t\t\t");
    Serial.println(i);
    settings.gain = i;
    Pozyx.setUWBSettings(settings);
    Pozyx.getUWBSettings(&settings);
    Serial.print("getUWBSettings.gain:\t\t");
    Serial.println(settings.gain, DEC);
    Pozyx.getTxPower(&gain);
    Serial.print("getTxPower:\t\t\t");
    Serial.println(gain, DEC);
    Serial.println();
  }
}

void loop()
{
  delay(500);
}
```
Results in:

> Pozyx Shield
> ------------
> WhoAmI: 0x43
> SW ver.: 1
> HW ver.: 35
> selftest: 0b111111
> 
> Default values:
> getUWBSettings.gain: 23
> getTxPower: 11.5000000000
> 
> Starting setTxPower test:
> setTxPower:			0
> getUWBSettings.gain:		0
> getTxPower:			0.0000000000
> 
> setTxPower:			1
> getUWBSettings.gain:		2
> getTxPower:			1.0000000000
> 
> setTxPower:			2
> getUWBSettings.gain:		4
> getTxPower:			2.0000000000
> 
> setTxPower:			3
> getUWBSettings.gain:		4
> getTxPower:			3.0000000000
> 
> setTxPower:			4
> getUWBSettings.gain:		6
> getTxPower:			4.0000000000
> 
> setTxPower:			5
> getUWBSettings.gain:		8
> getTxPower:			5.0000000000
> 
> setTxPower:			6
> getUWBSettings.gain:		10
> getTxPower:			6.0000000000
> 
> setTxPower:			7
> getUWBSettings.gain:		14
> getTxPower:			7.0000000000
> 
> setTxPower:			8
> getUWBSettings.gain:		16
> getTxPower:			8.0000000000
> 
> setTxPower:			9
> getUWBSettings.gain:		18
> getTxPower:			9.0000000000
> 
> setTxPower:			10
> getUWBSettings.gain:		18
> getTxPower:			10.0000000000
> 
> setTxPower:			11
> getUWBSettings.gain:		22
> getTxPower:			11.0000000000
> 
> setTxPower:			12
> getUWBSettings.gain:		24
> getTxPower:			12.0000000000
> 
> setTxPower:			13
> getUWBSettings.gain:		24
> getTxPower:			13.0000000000
> 
> setTxPower:			14
> getUWBSettings.gain:		28
> getTxPower:			14.0000000000
> 
> setTxPower:			15
> getUWBSettings.gain:		28
> getTxPower:			15.0000000000
> 
> setTxPower:			16
> getUWBSettings.gain:		32
> getTxPower:			16.0000000000
> 
> setTxPower:			17
> getUWBSettings.gain:		34
> getTxPower:			17.0000000000
> 
> setTxPower:			18
> getUWBSettings.gain:		34
> getTxPower:			18.0000000000
> 
> setTxPower:			19
> getUWBSettings.gain:		38
> getTxPower:			19.0000000000
> 
> setTxPower:			20
> getUWBSettings.gain:		40
> getTxPower:			20.0000000000
> 
> setTxPower:			21
> getUWBSettings.gain:		40
> getTxPower:			21.0000000000
> 
> setTxPower:			22
> getUWBSettings.gain:		44
> getTxPower:			22.0000000000
> 
> setTxPower:			23
> getUWBSettings.gain:		44
> getTxPower:			23.0000000000
> 
> setTxPower:			24
> getUWBSettings.gain:		46
> getTxPower:			24.0000000000
> 
> setTxPower:			25
> getUWBSettings.gain:		50
> getTxPower:			25.0000000000
> 
> setTxPower:			26
> getUWBSettings.gain:		50
> getTxPower:			26.0000000000
> 
> setTxPower:			27
> getUWBSettings.gain:		54
> getTxPower:			27.0000000000
> 
> setTxPower:			28
> getUWBSettings.gain:		56
> getTxPower:			28.0000000000
> 
> setTxPower:			29
> getUWBSettings.gain:		56
> getTxPower:			29.0000000000
> 
> setTxPower:			30
> getUWBSettings.gain:		60
> getTxPower:			30.0000000000
> 
> setTxPower:			31
> getUWBSettings.gain:		62
> getTxPower:			31.0000000000
> 
> setTxPower:			32
> getUWBSettings.gain:		62
> getTxPower:			32.0000000000
> 
> setTxPower:			**33**
> getUWBSettings.gain:		**66**
> getTxPower:			**33.0000000000**
> 
> setTxPower:			**34**
> getUWBSettings.gain:		**66**
> getTxPower:			**33.0000000000**
> 
> setTxPower:			**35**
> getUWBSettings.gain:		**66**
> getTxPower:			**33.0000000000**
> 
> 
> 
> Starting setUWBSettings test:
> setUWBSettings:			0
> getUWBSettings.gain:		0
> getTxPower:			0.0000000000
> 
> setUWBSettings:			1
> getUWBSettings.gain:		1
> getTxPower:			0.5000000000
> 
> setUWBSettings:			2
> getUWBSettings.gain:		2
> getTxPower:			1.0000000000
> 
> setUWBSettings:			3
> getUWBSettings.gain:		3
> getTxPower:			1.5000000000
> 
> setUWBSettings:			4
> getUWBSettings.gain:		4
> getTxPower:			2.0000000000
> 
> setUWBSettings:			5
> getUWBSettings.gain:		5
> getTxPower:			2.5000000000
> 
> setUWBSettings:			6
> getUWBSettings.gain:		6
> getTxPower:			3.0000000000
> 
> setUWBSettings:			7
> getUWBSettings.gain:		7
> getTxPower:			3.5000000000
> 
> setUWBSettings:			8
> getUWBSettings.gain:		8
> getTxPower:			4.0000000000
> 
> setUWBSettings:			9
> getUWBSettings.gain:		9
> getTxPower:			4.5000000000
> 
> setUWBSettings:			10
> getUWBSettings.gain:		10
> getTxPower:			5.0000000000
> 
> setUWBSettings:			11
> getUWBSettings.gain:		11
> getTxPower:			5.5000000000
> 
> setUWBSettings:			12
> getUWBSettings.gain:		12
> getTxPower:			6.0000000000
> 
> setUWBSettings:			13
> getUWBSettings.gain:		13
> getTxPower:			6.5000000000
> 
> setUWBSettings:			14
> getUWBSettings.gain:		14
> getTxPower:			7.0000000000
> 
> setUWBSettings:			15
> getUWBSettings.gain:		15
> getTxPower:			7.5000000000
> 
> setUWBSettings:			16
> getUWBSettings.gain:		16
> getTxPower:			8.0000000000
> 
> setUWBSettings:			17
> getUWBSettings.gain:		17
> getTxPower:			8.5000000000
> 
> setUWBSettings:			18
> getUWBSettings.gain:		18
> getTxPower:			9.0000000000
> 
> setUWBSettings:			19
> getUWBSettings.gain:		19
> getTxPower:			9.5000000000
> 
> setUWBSettings:			20
> getUWBSettings.gain:		20
> getTxPower:			10.0000000000
> 
> setUWBSettings:			21
> getUWBSettings.gain:		21
> getTxPower:			10.5000000000
> 
> setUWBSettings:			22
> getUWBSettings.gain:		22
> getTxPower:			11.0000000000
> 
> setUWBSettings:			23
> getUWBSettings.gain:		23
> getTxPower:			11.5000000000
> 
> setUWBSettings:			24
> getUWBSettings.gain:		24
> getTxPower:			12.0000000000
> 
> setUWBSettings:			25
> getUWBSettings.gain:		25
> getTxPower:			12.5000000000
> 
> setUWBSettings:			26
> getUWBSettings.gain:		26
> getTxPower:			13.0000000000
> 
> setUWBSettings:			27
> getUWBSettings.gain:		27
> getTxPower:			13.5000000000
> 
> setUWBSettings:			28
> getUWBSettings.gain:		28
> getTxPower:			14.0000000000
> 
> setUWBSettings:			29
> getUWBSettings.gain:		29
> getTxPower:			14.5000000000
> 
> setUWBSettings:			30
> getUWBSettings.gain:		30
> getTxPower:			15.0000000000
> 
> setUWBSettings:			31
> getUWBSettings.gain:		31
> getTxPower:			15.5000000000
> 
> setUWBSettings:			32
> getUWBSettings.gain:		32
> getTxPower:			16.0000000000
> 
> setUWBSettings:			33
> getUWBSettings.gain:		33
> getTxPower:			16.5000000000
> 
> setUWBSettings:			34
> getUWBSettings.gain:		34
> getTxPower:			17.0000000000
> 
> setUWBSettings:			35
> getUWBSettings.gain:		35
> getTxPower:			17.5000000000
> 
> setUWBSettings:			36
> getUWBSettings.gain:		36
> getTxPower:			18.0000000000
> 
> setUWBSettings:			37
> getUWBSettings.gain:		37
> getTxPower:			18.5000000000
> 
> setUWBSettings:			38
> getUWBSettings.gain:		38
> getTxPower:			19.0000000000
> 
> setUWBSettings:			39
> getUWBSettings.gain:		39
> getTxPower:			19.5000000000
> 
> setUWBSettings:			40
> getUWBSettings.gain:		40
> getTxPower:			20.0000000000
> 
> setUWBSettings:			41
> getUWBSettings.gain:		41
> getTxPower:			20.5000000000
> 
> setUWBSettings:			42
> getUWBSettings.gain:		42
> getTxPower:			21.0000000000
> 
> setUWBSettings:			43
> getUWBSettings.gain:		43
> getTxPower:			21.5000000000
> 
> setUWBSettings:			44
> getUWBSettings.gain:		44
> getTxPower:			22.0000000000
> 
> setUWBSettings:			45
> getUWBSettings.gain:		45
> getTxPower:			22.5000000000
> 
> setUWBSettings:			46
> getUWBSettings.gain:		46
> getTxPower:			23.0000000000
> 
> setUWBSettings:			47
> getUWBSettings.gain:		47
> getTxPower:			23.5000000000
> 
> setUWBSettings:			48
> getUWBSettings.gain:		48
> getTxPower:			24.0000000000
> 
> setUWBSettings:			49
> getUWBSettings.gain:		49
> getTxPower:			24.5000000000
> 
> setUWBSettings:			50
> getUWBSettings.gain:		50
> getTxPower:			25.0000000000
> 
> setUWBSettings:			51
> getUWBSettings.gain:		51
> getTxPower:			25.5000000000
> 
> setUWBSettings:			52
> getUWBSettings.gain:		52
> getTxPower:			26.0000000000
> 
> setUWBSettings:			53
> getUWBSettings.gain:		53
> getTxPower:			26.5000000000
> 
> setUWBSettings:			54
> getUWBSettings.gain:		54
> getTxPower:			27.0000000000
> 
> setUWBSettings:			55
> getUWBSettings.gain:		55
> getTxPower:			27.5000000000
> 
> setUWBSettings:			56
> getUWBSettings.gain:		56
> getTxPower:			28.0000000000
> 
> setUWBSettings:			57
> getUWBSettings.gain:		57
> getTxPower:			28.5000000000
> 
> setUWBSettings:			58
> getUWBSettings.gain:		58
> getTxPower:			29.0000000000
> 
> setUWBSettings:			59
> getUWBSettings.gain:		59
> getTxPower:			29.5000000000
> 
> setUWBSettings:			60
> getUWBSettings.gain:		60
> getTxPower:			30.0000000000
> 
> setUWBSettings:			61
> getUWBSettings.gain:		61
> getTxPower:			30.5000000000
> 
> setUWBSettings:			62
> getUWBSettings.gain:		62
> getTxPower:			31.0000000000
> 
> setUWBSettings:			63
> getUWBSettings.gain:		63
> getTxPower:			31.5000000000
> 
> setUWBSettings:			64
> getUWBSettings.gain:		64
> getTxPower:			32.0000000000
> 
> setUWBSettings:			65
> getUWBSettings.gain:		65
> getTxPower:			32.5000000000
> 
> setUWBSettings:			66
> getUWBSettings.gain:		66
> getTxPower:			33.0000000000
> 
> setUWBSettings:			67
> getUWBSettings.gain:		67
> getTxPower:			33.5000000000
> 
> setUWBSettings:			68
> getUWBSettings.gain:		23
> getTxPower:			**11.5000000000**
> 
> setUWBSettings:			69
> getUWBSettings.gain:		23
> getTxPower:			**11.5000000000**
> 
> setUWBSettings:			70
> getUWBSettings.gain:		23
> getTxPower:			**11.5000000000**

Basically the two functions don't do the same thing, in different resolutions, and neither can set the max.

**Comment by samuelvdv** [comment]
The maximum TX gain value is in fact 33.5dB, and not 35dB as is stated in the documentation. 
Also, immediately reading the TX power after setting it doesn't give the pozyx device the time to properly reconfigure the power, the new value is only 'written' after the transmit power is successfully reconfigured. To accommodate for this, I have added a 1ms delay in the ```setTxPower``` function.

Also, to make the library more consistent, I have changed the UWB_settings structure to contain the gain in dB, and removed the trim value, as this value should remain fixed anyhow. This change also involved a change to the functions ```getUWBSettings``` and ```setUWBSettings```. 
Note that it is known that currently, the ```setUWBSettings``` cannot successfully configure every one combination of the UWB parameters, we are testing this and fixing this in the next firmware update.

Below I have edited your sketch to show that it now all works:

```C
#include <Pozyx.h>
#include <Pozyx_definitions.h>
#include <Wire.h>

void setup()
{
  int status;
  float gain = 0;
  UWB_settings_t settings, settings2;

  Serial.begin(115200);

  Pozyx.begin(true);
  
  // Small test for resetting the UWB parameters
  settings2.channel = 2;
  settings2.plen = 0x34;
  settings2.bitrate = 2;
  settings2.prf = 1;
  settings2.gain_db = 10.0f;  
  Pozyx.setUWBSettings(&settings2);
  
  Serial.println("\nUWB parameters:");
  Pozyx.getUWBSettings(&settings);
  Serial.print("channel:\t\t");
  Serial.println(settings.channel);
  Serial.print("PRF:\t\t\t");
  Serial.println(settings.prf, HEX);
  Serial.print("bitrate:\t\t");
  Serial.println(settings.bitrate);
  Serial.print("preamble length:\t");
  Serial.println(settings.plen);
  Serial.print("gain (dB): \t\t");
  Serial.println(settings.gain_db);
      
  Serial.println("\n----------------------------------");      
  Serial.println("\nStarting setTxPower test:");  
  for(int i = 0; i < 68; i++)
  {
    Serial.print("setTxPower:\t\t\t");
    Serial.println(i*0.5f);
    Pozyx.setTxPower(i*0.5f);
    Pozyx.getUWBSettings(&settings);
    Serial.print("getUWBSettings.gain:\t\t");
    Serial.println(settings.gain_db);
    Pozyx.getTxPower(&gain);
    Serial.print("getTxPower:\t\t\t");
    Serial.println(gain);
    Serial.println();
  }  
}

void loop()
{
  delay(500);
}
```

The output is as follows:
```
Pozyx Shield
------------
WhoAmI: 0x43
SW ver.: 1
HW ver.: 35
selftest: 0b111111
2
10.00

UWB parameters:
channel:		2
PRF:			1
bitrate:		2
preamble length:	52
gain (dB): 		10.00

----------------------------------

Starting setTxPower test:
setTxPower:			0.00
getUWBSettings.gain:		0.00
getTxPower:			0.00

setTxPower:			0.50
getUWBSettings.gain:		0.50
getTxPower:			0.50

setTxPower:			1.00
getUWBSettings.gain:		1.00
getTxPower:			1.00

setTxPower:			1.50
getUWBSettings.gain:		1.50
getTxPower:			1.50

setTxPower:			2.00
getUWBSettings.gain:		2.00
getTxPower:			2.00

setTxPower:			2.50
getUWBSettings.gain:		2.50
getTxPower:			2.50

setTxPower:			3.00
getUWBSettings.gain:		3.00
getTxPower:			3.00

setTxPower:			3.50
getUWBSettings.gain:		3.50
getTxPower:			3.50

setTxPower:			4.00
getUWBSettings.gain:		4.00
getTxPower:			4.00

setTxPower:			4.50
getUWBSettings.gain:		4.50
getTxPower:			4.50

setTxPower:			5.00
getUWBSettings.gain:		5.00
getTxPower:			5.00

setTxPower:			5.50
getUWBSettings.gain:		5.50
getTxPower:			5.50

setTxPower:			6.00
getUWBSettings.gain:		6.00
getTxPower:			6.00

setTxPower:			6.50
getUWBSettings.gain:		6.50
getTxPower:			6.50

setTxPower:			7.00
getUWBSettings.gain:		7.00
getTxPower:			7.00

setTxPower:			7.50
getUWBSettings.gain:		7.50
getTxPower:			7.50

setTxPower:			8.00
getUWBSettings.gain:		8.00
getTxPower:			8.00

setTxPower:			8.50
getUWBSettings.gain:		8.50
getTxPower:			8.50

setTxPower:			9.00
getUWBSettings.gain:		9.00
getTxPower:			9.00

setTxPower:			9.50
getUWBSettings.gain:		9.50
getTxPower:			9.50

setTxPower:			10.00
getUWBSettings.gain:		10.00
getTxPower:			10.00

setTxPower:			10.50
getUWBSettings.gain:		10.50
getTxPower:			10.50

setTxPower:			11.00
getUWBSettings.gain:		11.00
getTxPower:			11.00

setTxPower:			11.50
getUWBSettings.gain:		11.50
getTxPower:			11.50

setTxPower:			12.00
getUWBSettings.gain:		12.00
getTxPower:			12.00

setTxPower:			12.50
getUWBSettings.gain:		12.50
getTxPower:			12.50

setTxPower:			13.00
getUWBSettings.gain:		13.00
getTxPower:			13.00

setTxPower:			13.50
getUWBSettings.gain:		13.50
getTxPower:			13.50

setTxPower:			14.00
getUWBSettings.gain:		14.00
getTxPower:			14.00

setTxPower:			14.50
getUWBSettings.gain:		14.50
getTxPower:			14.50

setTxPower:			15.00
getUWBSettings.gain:		15.00
getTxPower:			15.00

setTxPower:			15.50
getUWBSettings.gain:		15.50
getTxPower:			15.50

setTxPower:			16.00
getUWBSettings.gain:		16.00
getTxPower:			16.00

setTxPower:			16.50
getUWBSettings.gain:		16.50
getTxPower:			16.50

setTxPower:			17.00
getUWBSettings.gain:		17.00
getTxPower:			17.00

setTxPower:			17.50
getUWBSettings.gain:		17.50
getTxPower:			17.50

setTxPower:			18.00
getUWBSettings.gain:		18.00
getTxPower:			18.00

setTxPower:			18.50
getUWBSettings.gain:		18.50
getTxPower:			18.50

setTxPower:			19.00
getUWBSettings.gain:		19.00
getTxPower:			19.00

setTxPower:			19.50
getUWBSettings.gain:		19.50
getTxPower:			19.50

setTxPower:			20.00
getUWBSettings.gain:		20.00
getTxPower:			20.00

setTxPower:			20.50
getUWBSettings.gain:		20.50
getTxPower:			20.50

setTxPower:			21.00
getUWBSettings.gain:		21.00
getTxPower:			21.00

setTxPower:			21.50
getUWBSettings.gain:		21.50
getTxPower:			21.50

setTxPower:			22.00
getUWBSettings.gain:		22.00
getTxPower:			22.00

setTxPower:			22.50
getUWBSettings.gain:		22.50
getTxPower:			22.50

setTxPower:			23.00
getUWBSettings.gain:		23.00
getTxPower:			23.00

setTxPower:			23.50
getUWBSettings.gain:		23.50
getTxPower:			23.50

setTxPower:			24.00
getUWBSettings.gain:		24.00
getTxPower:			24.00

setTxPower:			24.50
getUWBSettings.gain:		24.50
getTxPower:			24.50

setTxPower:			25.00
getUWBSettings.gain:		25.00
getTxPower:			25.00

setTxPower:			25.50
getUWBSettings.gain:		25.50
getTxPower:			25.50

setTxPower:			26.00
getUWBSettings.gain:		26.00
getTxPower:			26.00

setTxPower:			26.50
getUWBSettings.gain:		26.50
getTxPower:			26.50

setTxPower:			27.00
getUWBSettings.gain:		27.00
getTxPower:			27.00

setTxPower:			27.50
getUWBSettings.gain:		27.50
getTxPower:			27.50

setTxPower:			28.00
getUWBSettings.gain:		28.00
getTxPower:			28.00

setTxPower:			28.50
getUWBSettings.gain:		28.50
getTxPower:			28.50

setTxPower:			29.00
getUWBSettings.gain:		29.00
getTxPower:			29.00

setTxPower:			29.50
getUWBSettings.gain:		29.50
getTxPower:			29.50

setTxPower:			30.00
getUWBSettings.gain:		30.00
getTxPower:			30.00

setTxPower:			30.50
getUWBSettings.gain:		30.50
getTxPower:			30.50

setTxPower:			31.00
getUWBSettings.gain:		31.00
getTxPower:			31.00

setTxPower:			31.50
getUWBSettings.gain:		31.50
getTxPower:			31.50

setTxPower:			32.00
getUWBSettings.gain:		32.00
getTxPower:			32.00

setTxPower:			32.50
getUWBSettings.gain:		32.50
getTxPower:			32.50

setTxPower:			33.00
getUWBSettings.gain:		33.00
getTxPower:			33.00

setTxPower:			33.50
getUWBSettings.gain:		33.50
getTxPower:			33.50

```
